src/domains/transaction/hooks/use-transaction-voting-wallets.ts

```diff
@@ -1,14 +1,31 @@
 import { useEnvironmentContext } from "@/app/contexts";
-import { Networks } from "@ardenthq/sdk";
-import { Contracts, DTO } from "@ardenthq/sdk-profiles";
+import { Networks, DTO } from "@ardenthq/sdk";
+import { Contracts } from "@ardenthq/sdk-profiles";
 import { useEffect, useState } from "react";
 
 interface Properties {
 	network: Networks.Network;
-	transaction: DTO.ExtendedConfirmedTransactionData;
+	transaction: DTO.RawTransactionData;
 	profile: Contracts.IProfile;
 }
 
+/**
+ * @TODO: This can be handled in SDK to retrieve voting and unvoting public keys (or generally the asset data)
+ * through a common method/format, whether it's a signed or confirmed transaction.
+ * Currently `data` is a function in signed transaction and an object in confirmed transaction.
+ */
+const extractVotingData = ({ transaction }: { transaction: DTO.RawTransactionData }) => {
+	const data =
+		typeof transaction.data?.().data === "function" ? transaction.data?.().data() : transaction.data?.().data;
+	const votes = data?.asset?.votes ?? [];
+	const unvotes = data?.asset?.unvotes ?? [];
```

Comment:
maybe turn this into two helpers, one for confirmed (using the `votes` and `unvotes` methods) and one for the signed tx that uses the raw asset? that will make it easier to follow why these checks are done and different data is accessed